usecases/backup/backupper.go

```diff
@@ -158,9 +158,17 @@ func (b *backupper) backup(ctx context.Context,
 			Version:       Version,
 			ServerVersion: config.ServerVersion,
 		}
-		if err := provider.all(context.Background(), req.Classes, &result); err != nil {
+
+		// the coordinator might want to abort the backup
+		done := make(chan struct{})
+		ctx := b.withCancellation(context.Background(), id, done)
```

Comment:
Same as above (avoid shadowing)

Reply:
Shadowing is exactly what we want here because the async Clojure defines its own context and has no use for the original context.